packages/react/src/components/Button/Button.tsx

```diff
@@ -172,11 +172,19 @@ const Button: React.FC<WithAsProp<ButtonProps>> &
   const unhandledProps = getUnhandledProps(Button.handledProps, props)
   const ElementType = getElementType(props)
 
+  const renderContent = () => {
+    // @ts-ignore TODO pls fix me
+    return ButtonContent.create(content, {
+      defaultProps: () => getA11Props('content', { as: 'span', size }),
+    })
+  }
+
   const renderIcon = () => {
-    return Icon.create(icon, {
+    // @ts-ignore
+    return ButtonIcon.create(icon, {
```

Comment:
Still not sure about the `ButtonIcon` component... Maybe in the end we can just combine once the styles for the Icon component + styles for the Button's icon slot and store that as a cache... I am worried that for more complicated component, creating variant for the component will mean creating new variants for each slot, and you may only need to change one style for them...